src/Units/UnitState.java

```diff
@@ -157,48 +155,27 @@ public int alterHP(int change, boolean allowOver)
       // Apply the cap as needed
       final int newHP = Math.min(capHP, oldHP + change);
       // Figure out whether that reduces our healing
-      realChange = Math.min(change, newHP - oldHP);
+      realChange = Math.min(change, newHP - health);
     }
 
-    health = Math.max(1, health + healthFromHP(realChange));
+    health = Math.max(1, health + realChange);
     // Round HP up, if healing
-    if( change >= 0 )
-      health = healthFromHP(getHP());
+    if( roundUp && change >= 0 )
+      health = getHP();
 
     return getHP() - oldHP;
   }
 
   /**
-   * Increases *fractional health* by the specified amount.
-   * <p>Enforces a minimum of 0.1, and a maximum (optional) of MAXIMUM_HP.
+   * Increases health by the specified amount.
+   * <p>Enforces a minimum of 1, and a maximum (optional) of MAXIMUM_HP.
    * <p>Does not round.
    * <p>Use this when you want precise non-combat health changes, or want to heal without rounding up.
    * @return the change in HP
    */
   public int alterHealthPercent(int percentChange)
```

Comment:
Is there any reason to have both `alterHP` and `alterHealthPercent`? I could see it if `alterHP` accepted and returned [1,10] and `alterHealthPercent` accepted and returned [1,100] maybe, but right now they are just clones.

Reply:
It's effectively an overload with different connotations, as is documented.
It doesn't round up HP when healing.
The single-bool overload of `alterHP()` allows overhealing, not the rounding toggle.

Reply:
Okay, I see the rationale for having both. I still feel like the naming could be tightened (maybe it's just a me problem that I can't look at `HP` and see [0-100]).
Also seems odd that it takes and applies a precise change, but returns a rounded value. Though I guess the rounded value is what always matters external to `Unit`.

Aside: Documentation here refers to an optional cap which isn't supported (no Boolean toggle param).

Reply:
Yeah. "HealthPercent" is a non-terse way to write it, which signals to me personally that it's an alternate function that does a non-expected thing (which, in this case, is not rounding up).
It's probably not fair to expect others to uh... read the tea leaves like that.

I've renamed it to [`alterHealthNoRound`](https://github.com/ThislsAUsername/DefendPeace/pull/401/commits/bbd41a202f5a35258c243b21d0e21630e3a1671a), and deleted the documentation since it looked like a restatement of the function name.